### PR TITLE
Change Python 2 scripts to Python 3 (SOFTWARE-4673)

### PR DESCRIPTION
**Pull request by mtru32**
urllib2 is now urllib and the requests are subsequently changed, I don't know if they work yet. Also updated the strings with '.format(**locals())' to f-strings

(Squash)
Added directory and changed import.
osgtest/vendor directory is from osg-test repo. stringToVersion is a function from a vendor so it is not subject to change from Python 2 to 3.

### CODE REVIEW

**Review by brianhlin — CHANGES_REQUESTED** [review]
Hey @mtru32 this is a bit of a big PR because a lot more of `osg-test` snuck in than I think we need. Could you do the following?

1. Remove the entirety of `osgtest/` from your branch and commit it
2. Do a `git rebase` to fixup the new commit from (1) into 263c3f6 
3. Create a `vendor/` dir in the root of this repo and copy only the contents of  https://github.com/opensciencegrid/osg-test/tree/master/osgtest/vendor into it, then commit that
4. Force push to automatically update this PR

File: dockerhub-prune-tags.py [hunk 1]
```diff
@@ -71,33 +71,37 @@ def authstr(user, passwd):
     from base64 import encodestring
     return encodestring('%s:%s' % (user,passwd)).replace('\n', '')
 
+
+## unsure about urllib request and response. It was originally
+# 'urllib2.request(url)' and 'urllib2.urlopen(req)'
```
Comment (edquist) [inline dockerhub-prune-tags.py]: Well, it was

```suggestion
# 'urllib2.Request(url)' and 'urllib2.urlopen(req)'
```

I think the "upgrade" from `urllib2` -> `urllib.request` is correct for these, but you need to `import urllib.request` explicitly.

I would appreciate you testing these to the point that you are no longer "unsure" about it though, and then removing these two comment lines!  :)

File: dockerhub-prune-tags.py [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 import collections
 import getpass
-import urllib2
+import urllib
```
Comment (edquist) [inline dockerhub-prune-tags.py]:
```suggestion
import urllib.request
```

File: dockerhub-prune-tags.py [hunk 3]
```diff
@@ -13,7 +13,7 @@
 
 def usage():
     script = os.path.basename(__file__)
-    print("Usage: {script} [opensciencegrid/]<REPO>".format(**locals()))
+    print(f"Usage: {script} [opensciencegrid/]<REPO>")
```
Comment (edquist) [inline dockerhub-prune-tags.py:16]: Hey now, `.format(**locals())` is legit python3!  :)

**Review by edquist — COMMENTED** [review]
In addition to the comments i've made, you'll want to update the shebang line to `#!/usr/bin/python3`, as this script will no longer work against python2, which /usr/bin/python often still is.

**Comment by mtru32** [comment]
@brianhlin

File: dockerhub-prune-tags.py [hunk 4]
```diff
@@ -68,36 +68,38 @@ def set_filter(fn, seq):
     return set(filter(fn, seq))
 
 def authstr(user, passwd):
-    from base64 import encodestring
-    return encodestring('%s:%s' % (user,passwd)).replace('\n', '')
+    from base64 import b64encode, encode
+    auth = f'{user}:{passwd}'
+    auth_bytes = p.encode()
+    return str(b64encode(auth_bytes))
```
Comment (edquist) [inline dockerhub-prune-tags.py]: Hi @mtru32,

A few things here:

- that is cool that `base64.b64encode` does not append a trailing newline
- it doesn't look like `encode` is used anywhere
- it doesn't look like `p` is _defined_ anywhere (won't even compile as a result)
- converting a `bytes` object with `str(...)` will give the `b'...'`, which is not what we want - you've gotta decode it to `str`

I think we can fix these with

```suggestion
    from base64 import b64encode
    auth = f'{user}:{passwd}'
    auth_bytes = auth.encode()
    return b64encode(auth_bytes).decode()
```

**Review by edquist — CHANGES_REQUESTED** [review]

File: dockerhub-prune-tags.py (hunk 3 above)
Comment (mtru32) [reply dockerhub-prune-tags.py:16]: My apologies :)

File: dockerhub-prune-tags.py (hunk 4 above)
Comment (mtru32) [reply dockerhub-prune-tags.py]: Hey @edquist thanks for the comments. My mistake on the 'p' variable, that was the result of renaming too quickly. Let me know if you need anything else!

**Review by edquist — APPROVED** [review]
I think that was it for me

**Review by brianhlin — APPROVED** [review]

**Pull request merged**